pkg/resources/cloudconfig/configmap.go

```diff
@@ -144,6 +145,7 @@ func CloudConfig(
 			},
 			LoadBalancer: openstack.LoadBalancerOpts{
 				ManageSecurityGroups: manageSecurityGroups == nil || *manageSecurityGroups,
+				UseOctavia: useOctavia == nil || *useOctavia,
```

Comment:
```suggestion
				UseOctavia: useOctavia,
```

`UseOctavia` expects a bool pointer ;-). You can avoid declaring `useOctavia` and use `dc.Spec.Openstack.UseOctavia` directly.

Reply:
I see, I updated the code accordingly. I am not sure if it makes sense to have
```
	ManageSecurityGroups bool         `gcfg:"manage-security-groups"`
	UseOctavia           *bool        `gcfg:"use-octavia"`
```
in the machine-controller. See: https://github.com/kubermatic/machine-controller/blob/af516ec83c7f27c930462ea708275cc58294cb57/pkg/cloudprovider/provider/openstack/types/cloudconfig.go#L60
Mainly, `ManageSecurityGroups` is a bool and `UseOctavia` is a boolptr?

Reply:
> Mainly, ManageSecurityGroups is a bool and UseOctavia is a boolptr?

The reason why UseOctavia is a pointer is that for OpenStack we still have old clusters using the in-tree cloud provider and newer ones using CCM and CSI. In the first case "use-octavia" is `false` by default, in the second is `true`. Having a pointer gives the possibility to omit it and keep using the respective default unless explicitly specified.
I should have put a commit to make this explicit, thx for pointing out.

Reply:
Thanks for the explanation! That makes sense and I wasn't aware of that. 👍 Good point! :)